src/riak_core_ring.erl

```diff
@@ -1367,17 +1440,36 @@ internal_ring_changed(Node, CState0) ->
     end.
 
 %% @private
-merge_meta(M1,M2) ->
-    dict:merge(fun(_,D1,D2) -> pick_val(D1,D2) end, M1, M2).
+merge_meta({N1,M1}, {N2,M2}) ->
+    Meta = dict:merge(fun(_,D1,D2) -> pick_val({N1,D1}, {N2,D2}) end, M1, M2),
+    log_meta_merge(M1, M2, Meta),
+    Meta.
 
 %% @private
-pick_val(M1,M2) ->
-    case M1#meta_entry.lastmod > M2#meta_entry.lastmod of
+pick_val({N1,M1}, {N2,M2}) ->
+    case {M1#meta_entry.lastmod, N1} > {M2#meta_entry.lastmod, N2} of
```

Comment:
Does this refer to the deterministic resolution in the comment?  Basically, add the node as a tie breaker?